utility/src/main/java/org/oppia/android/util/logging/CurrentAppScreenNameIntentDecorator.kt

```diff
@@ -0,0 +1,33 @@
+package org.oppia.android.util.logging
+
+import android.content.Intent
+import org.oppia.android.app.model.CurrentAppScreen
+import org.oppia.android.app.model.ScreenName
+import org.oppia.android.util.extensions.getProtoExtra
+import org.oppia.android.util.extensions.putProtoExtra
+
+private var CURRENT_APP_SCREEN_NAME_KEY = "CURRENT_APP_SCREEN_NAME_KEY"
+
+/** Utility to provide utilities related to wrapping the current app-screen name in an intent. */
```

Comment:
```suggestion
/** Decorator for wrapping an activity's [ScreenName] within its intent. */
```

Suggested changes for clarity and conciseness.

Reply:
done.